test/unit_tests/conversation/ConversationServiceSpec.coffee

```diff
@@ -140,61 +140,64 @@ describe 'Conversation Service', ->
-          object: {"raw":{"from":"8b497692-7a38-4a5d-8287-e3d1006577d6","time":"2016-08-04T13:28:30.019Z","type":"conversation.otr-message-add","conversation":"35a9a89d-70dc-4d9e-88a2-4d8758458a6a"},"meta":{"timestamp":1470317310019,"version":1},"mapped":{"conversation":"35a9a89d-70dc-4d9e-88a2-4d8758458a6a","id":"0880f7b9-b8f1-45e4-825e-fa120daa98b2","from":"8b497692-7a38-4a5d-8287-e3d1006577d6","time":"2016-08-04T13:28:30.019Z","data":{"content":"Fourth message","nonce":"0880f7b9-b8f1-45e4-825e-fa120daa98b2","previews":[]},"type":"conversation.message-add"}}
-        },
-        {
-          key: "#{conversation_id}@#{sender_id}@1470317313389"
-          object: {"raw":{"from":"8b497692-7a38-4a5d-8287-e3d1006577d6","time":"2016-08-04T13:28:33.389Z","type":"conversation.otr-message-add","conversation":"35a9a89d-70dc-4d9e-88a2-4d8758458a6a"},"meta":{"timestamp":1470317313389,"version":1},"mapped":{"conversation":"35a9a89d-70dc-4d9e-88a2-4d8758458a6a","id":"5a8cd79a-82bb-49ca-a59e-9a8e76df77fb","from":"8b497692-7a38-4a5d-8287-e3d1006577d6","time":"2016-08-04T13:28:33.389Z","data":{"content":"Fifth message","nonce":"5a8cd79a-82bb-49ca-a59e-9a8e76df77fb","previews":[]},"type":"conversation.message-add"}}
-        }
-      ]
-      # @formatter:on
+    it 'loads all events with limit', (done) ->
+      conversation_service.load_events_from_db conversation_id, null, null, 5
+      .then (events) =>
+        expect(events.length).toBe 5
+        done()
 
-      promises = []
-      for message in messages
-        promise = conversation_service.storage_service.save storage_service.OBJECT_STORE_CONVERSATION_EVENTS, message.key, message.object
-        promises.push promise
-
-      Promise.all promises
-      .then =>
-        limit = 4
-        conversation_service.load_events_from_db conversation_id, undefined, limit
-        .then (loaded_events) =>
-          [events, has_further_events] = loaded_events
-          expect(events.length).toBe limit
-          expect(has_further_events).toBe true
-          expect(events[0].meta.timestamp).toBe messages[messages.length - 1].object.meta.timestamp
-          expect(events[1].meta.timestamp).toBe messages[messages.length - 2].object.meta.timestamp
-          expect(events[2].meta.timestamp).toBe messages[messages.length - 3].object.meta.timestamp
-          expect(events[3].meta.timestamp).toBe messages[messages.length - limit].object.meta.timestamp
-          done()
+    it 'loads events with start timestamp', (done) ->
+      conversation_service.load_events_from_db conversation_id, 4
+      .then (events) =>
+        expect(events.length).toBe 4
+        expect(events[0].meta.timestamp).toBe 3
+        expect(events[1].meta.timestamp).toBe 2
+        expect(events[2].meta.timestamp).toBe 1
+        expect(events[3].meta.timestamp).toBe 0
+        done()
+
+    it 'loads events with start and end timestamp', (done) ->
+      conversation_service.load_events_from_db conversation_id, 8, 6
```

Comment:
In `@param` of `load_events_from_db`, `start` and `end` are `Strings` but in the test case you use numbers.